### PR TITLE
LPAL-336 - Profile session management in front-end app

### PR DESCRIPTION
**Pull request by Raffers**
## Purpose

https://opgtransform.atlassian.net/browse/LPAL-336?atlOrigin=eyJpIjoiMGJjZDk3ODdlYmIwNGNlZTg1MjkwMTVmZjY4MmFhZGEiLCJwIjoiaiJ9

Fixes LPAL-336

## Approach

Replaces DynamoDB with Redis, no additional tests have been added. Existing tests have been updated to reflect changes

## Learning

_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the LPA service_

## Checklist

* [x] I have performed a self-review of my own code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [x] I have added tests to prove my work
* [x] I have added mandatory tags to terraformed resources, where possible
* [ ] If I added a package.json or composer.json, I also made sure this is included in the script in `.github/workflows/dependabot-update.yml`
* [ ] The product team have tested these changes

### CODE REVIEW

File: terraform/environment/ecs_front.tf [hunk 1]
```diff
@@ -233,7 +233,8 @@ locals {
         { "name" : "OPG_LPA_COMMON_PDF_CACHE_S3_BUCKET", "value" : data.aws_s3_bucket.lpa_pdf_cache.bucket },
         { "name" : "OPG_LPA_COMMON_PDF_QUEUE_URL", "value" : aws_sqs_queue.pdf_fifo_queue.id },
         { "name" : "OPG_LPA_ENDPOINTS_API", "value" : "http://${local.api_service_fqdn}" },
-        { "name" : "OPG_LPA_OS_PLACES_HUB_ENDPOINT", "value" : "https://api.os.uk/search/places/v1/postcode" }
+        { "name" : "OPG_LPA_OS_PLACES_HUB_ENDPOINT", "value" : "https://api.os.uk/search/places/v1/postcode" },
+        { "name" : "OPG_LPA_COMMON_REDIS_CACHE_URL", "value" : data.aws_elasticache_cluster.redis_cache.cache_nodes.0.address}
```
Comment (williamfalconeruk) [inline terraform/environment/ecs_front.tf:237]:
```suggestion
        { "name" : "OPG_LPA_COMMON_REDIS_CACHE_URL", "value" : "tls://${data.aws_elasticache_cluster.redis_cache.primary_endpoint_address}"
```
similar to this line: https://github.com/ministryofjustice/opg-use-an-lpa/blob/0ecc79f7de30c9e8ef3bc4ce5280cb48780fb02b/terraform/environment/actor_ecs.tf#L231

File: terraform/environment/shared_data_sources.tf [hunk 2]
```diff
@@ -0,0 +1,7 @@
+data "aws_security_group" "redis_cache_service" {
+  name = "redis-cache-service"
+}
+
+data "aws_elasticache_replication_group" "redis_cache" {
+  replication_group_id = "redis-cache-replication-group"
+}
```
Comment (williamfalconeruk) [inline terraform/environment/shared_data_sources.tf:7]: add a railing newline. might be worth running `terraform fmt` over your environment folder also...

File: service-front/docker/app/Dockerfile [hunk 3]
```diff
@@ -10,12 +10,13 @@ FROM php:7.4.16-fpm-alpine3.12
...
 
 RUN apk add --update --no-cache --virtual .build-dependencies $PHPIZE_DEPS icu-dev \
+        && pecl install redis \
         && pecl install xdebug \
         && pecl clear-cache \
+        && docker-php-ext-enable redis \
         && docker-php-ext-install intl \
         && docker-php-ext-install bcmath \
-        && docker-php-ext-install opcache \
-        && apk del .build-dependencies
```
Comment (williamfalconeruk) [inline service-front/docker/app/Dockerfile:18]: QQ: was there a technical reason the  the delete build dependancies line was removed?

**Review by williamfalconeruk — CHANGES_REQUESTED** [review]
Will approve once the one question is answered.

File: service-front/docker/app/Dockerfile (hunk 3 above)
Comment (Raffers) [reply service-front/docker/app/Dockerfile:18]: Mistake, will add that back in now.

**Review by williamfalconeruk — APPROVED** [review]

**Pull request merged**